src/ensembl/src/content/app/entity-viewer/shared/rest/rest-adaptors/rest-transcript-adaptor.ts

```diff
@@ -173,3 +93,56 @@ const calculateRelativeLocation = (
 ) => {
   return featurePosition - parentPosition; // not sure if this is correct
 };
+
+// transform ensembl rest /overlap data into a transcript data structure
+export const buildProduct = (
+  translation: TranslationInResponse,
+  proteinFeatures: ProteinFeature[]
+): Product => {
+  const protein_domains_resources = buildProteinDomainsResources(
+    proteinFeatures
+  );
+
+  return {
+    type: ProductType.PROTEIN,
+    stable_id: translation.id,
+    length: translation.length as number,
```

Comment:
why the `as number`?

Reply:
Ah that was because of some hacky code I had written initially. Will remove it.

Reply:
✔️